server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java

```diff
@@ -21,14 +21,19 @@ public Resp3AuthHandler(RespServer server) {
    }
 
    @Override
-   protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandlerContext ctx, RespCommand type, List<byte[]> arguments) {
-      if (type instanceof AuthResp3Command) {
-         AuthResp3Command type1 = (AuthResp3Command) type;
-         return type1.perform(this, ctx, arguments);
+   protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandlerContext ctx, RespCommand command, List<byte[]> arguments) {
+      if (command instanceof AuthResp3Command) {
+         if (command.hasValidNumberOfArguments(arguments)) {
```

Comment:
We can avoid the `hasValidNumberOfArguments` in each of the handlers and have it just on the `RespRequestHandler#handleRequest` before invoking the `actualHandleRequest`.

Reply:
I had a doubt on this one to be honest !